src/realm/sync/client.cpp

```diff
@@ -1458,6 +1482,11 @@ void SessionWrapper::on_suspended(const SessionErrorInfo& error_info)
 void SessionWrapper::on_resumed()
 {
     m_suspended = false;
+    if (m_flx_subscription_store && m_flx_pending_mark_version != SubscriptionSet::EmptyVersion) {
```

Comment:
Do we have a test case for this scenario?

Reply:
We didn't, so I went and wrote one, and then found out this block wasn't necessary and have removed it (but left the test case).